### PR TITLE
Add debug scripts for compiler passes

### PR DESCRIPTION
**Pull request by rengolin**
This is a simple set of scripts that run mlir-gen/tpp-opt, dumps the IR between passes, splits the output for each pass into files, looks at each one when the IR changes and runs the diff tool of your choice to show these differences.

For now, mlir-gen is only emitting the standard odd-shaped MLP layer over the standard TPP passes, but future changes to the scripts can add arguments to each tool.

### CODE REVIEW

**Comment by adam-smnk** [comment]
When I run `debug_all_passes.sh` it finishes with:
```
Traceback (most recent call last):
  File "/home/asiemien/plaidml-tpp-mlir/scripts/debug/diff.py", line 77, in <module>
    if sameFile(curr, next):
  File "/home/asiemien/plaidml-tpp-mlir/scripts/debug/diff.py", line 32, in sameFile
    rhs = readFile(next)
  File "/home/asiemien/plaidml-tpp-mlir/scripts/debug/diff.py", line 21, in readFile
    with open(file, "r") as fp:
TypeError: expected str, bytes or os.PathLike object, not NoneType
```

File: scripts/debug/split.py [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+
+    if not os.path.isfile(args.filename):
+        print(f"{args.filename} is not a file\n")
+        syntax()
+
+    ext=os.path.splitext(args.filename)[1]
+    counter = 0
+    filename = f"%03d{ext}" % counter
+    outfile = open(filename, "a")
+    with open(args.filename, "r") as file:
```
Comment (adam-smnk) [inline scripts/debug/split.py:26]: nit: could be combined into one `with` block to avoid explicit close

**Review by adam-smnk — APPROVED** [review]
Minus the mentioned error, looks nice 👍

File: scripts/debug/split.py (hunk 1 above)
Comment (rengolin) [reply scripts/debug/split.py:26]: It cannot, because the outfile is closed/opened inside the loop, it's not the same outfile, it's ALL outfiles.

Comment (adam-smnk) [reply scripts/debug/split.py]: Ah right, somehow missed that 👍

**Pull request merged**